app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,30 @@
+class CommentsController < ApplicationController
+  before_action :attr_user_post
+
+  def new
+    @comment = Comment.new(user: current_user, post: @post)
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.user_id = params[:user_id]
+    @comment.post_id = params[:post_id]
+    if @comment.save!
+      flash[:success] = ' Comment added succefully '
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def attr_user_post
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+  end
```

Comment:
- - Requirement screenshot:
![Selection_051](https://github.com/78fatemanazari/Interacti_Blog/assets/107632780/9a5df00b-0a37-4f16-be7b-37803c65abc5)


- [ ] According to the project requirement, the comment should be created onbehalf of the `current user` which you get by calling the `current_user` method you created. currently, you are getting the user by the ID passed from the request, although this works, that's not what was required. I suggest you stick to the requirement and use the `current_user` method here. :smile: :wave: